### PR TITLE
Implement Prototype

### PR DESCRIPTION
**Pull request by hayfield**
This adds a prototype that loads and parses data files, then outputs a multicoloured sortable and searchable table containing data.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,6 @@
+# untracked data
```
Comment (dalepotter) [inline .gitignore]: Small point, but could make the comment more descriptive - e.g. `# CSV and data files pulled from a remote source`

File: src/app.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+from flask import Flask
+from flask import render_template, url_for
+
+import parse_data
+
+app = Flask(__name__)
+
+
+@app.route("/hello")
```
Comment (dalepotter) [inline src/app.py]: This route could be deleted!

File: src/parse_data.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+
+def publisherify_data(base_info, summary_stats):
+	"""
+	Converts data to be in a format that allows direct querying by publisher.
+
+	Params:
+		base_info (array of dict): An array of dictionaries containing the base info.
+		summary_stats (array of dict): An array of dictionaries containing the summary stats.
+
```
Comment (dalepotter) [inline src/parse_data.py]: I would suggest being more precise about what type of array is expected in the input params - i.e. a list or set? I don't think the function would work if an associative array was used.

File: src/parse_data.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+	Returns:
+		A dictionary of dictionaries. The keys in the first-level dictionary are publisher registry IDs. Keys at the second level are names of statistics parsed from data file headers.
```
Comment (dalepotter) [inline src/parse_data.py]: Suggest split over multiple lines to reduce line length

File: src/parse_data.py [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+	return data
+
+
+def load_and_format_data():
+	"""
+	Loads and formats all data to be queried by publisher.
+
+	Returns:
+		A dictionary of dictionaries. The keys in the first-level dictionary are publisher registry IDs. Keys at the second level are names of statistics parsed from data file headers.
```
Comment (dalepotter) [inline src/parse_data.py]: Same comment as above re line length :-)

File: src/parse_data.py (hunk 3 above)
Comment (hayfield) [reply src/parse_data.py]: Is `list` specific enough?

**Review by dalepotter — CHANGES_REQUESTED** [review]
Could we fix up the docstrings as per the comments, particularity to clarify the comment re arrays before we merge?

**Comment by hayfield** [comment]
@dalepotter Changes made in response to comments.

**Comment by dalepotter** [comment]
👍  

Looks good to me, thanks for the changes :-)

**Review by dalepotter — APPROVED** [review]

**Pull request merged**